examples/apps/plotly_plots/app.py

```diff
@@ -0,0 +1,443 @@
+        'labels': ['Category A', 'Category B', 'Category C',
+                   'Sub A1', 'Sub A2', 'Sub B1', 'Sub B2', 'Sub C1'],
+        'parents': ['', '', '', 'A', 'A', 'B', 'B', 'C'],
+        'values': [0, 0, 0, 15, 25, 20, 10, 30]
+    })
+
+    fig = go.Figure(go.Sunburst(
+        ids=df['ids'],
+        labels=df['labels'],
+        parents=df['parents'],
+        values=df['values'],
+        branchvalues="total"
+    ))
+
+    fig.update_layout(
+        title="☀️ Sunburst Chart - Hierarchical Data",
+        font_size=12
+    )
+
+    return fig
+
+# Create reactive plot function that updates with theme changes
+def create_reactive_plots(dark_theme=False):
+    """Create all plots and update them based on theme."""
+    plots = [
+        create_line_plot(),
+        create_bar_chart(),
+        create_pie_chart(),
+        create_heatmap(),
+        create_surface_plot(),
+        create_candlestick(),
+        create_scatter_3d(),
+        create_parallel_coordinates(),
+        create_contour_plot(),
+        create_box_plot(),
+        create_histogram(),
+        create_sunburst()
+    ]
+
+    # Update all plots with the current theme
+    updated_plots = [update_template(plot, dark_theme=dark_theme) for plot in plots]
+
+    return updated_plots
+
+# Create the main dashboard function
+def create_dashboard():
+    """Create the main dashboard with Page component."""
+
+    # Create title and description
+    title = pmui.Typography("🎨 Panel Material UI Plotly Dashboard", variant="h3", sx={"mb": 2})
+    description = pmui.Typography(
+        "A comprehensive showcase of Plotly charts with Material Design theming. "
+        "Toggle between light and dark themes using the theme switcher in the header.",
+        variant="body1",
+        sx={"mb": 3, "color": "text.secondary"}
+    )
+
+    def plots_grid(dark_theme):
+        updated_plots = create_reactive_plots(dark_theme)
+
+        # Create grid layout with plots
+        plot_cards = []
+        for i, plot in enumerate(updated_plots):
+            card = pmui.Card(
+                pn.pane.Plotly(plot, sizing_mode='stretch_width', height=400),
+                sx={"mb": 2}
+            )
+            plot_cards.append(card)
+
+        # Arrange plots in a responsive grid
+        return pmui.Container(*[
+            pmui.Grid(*[
+                pmui.Grid(card, size={'xs': 12, 'md': 6, 'lg': 6}) for card in plot_cards
+            ], container=True, spacing=2, sx={"mb": 3}),
+        ])
+
+    page = pmui.Page(
+        title="Plotly Dashboard - Panel Material UI",
+        theme_toggle=True,
+        sx={
+            "& .main": {
+                "padding": "24px"
+            },
+            "& .sidebar": {
+                "padding": "24px"
+            }
+        },
+        theme_config=THEME_CONFIG,
+    )
+
+    page.main = [
+        title,
+        description,
+        pn.bind(plots_grid, page.param.dark_theme),
```

Comment:
I find it awkward that I cannot bind to the theme before I have a `Page` instance.

I think there should be a `pn.state.dark_theme` or similar parameter I could bind to.